test/mocks/http/mocks.cc

```diff
@@ -160,5 +165,94 @@ MockInstance::MockInstance() {}
 MockInstance::~MockInstance() {}
 
 } // namespace ConnectionPool
+
+bool HeaderValueOfMatcher::MatchAndExplain(const HeaderMap& headers,
+                                           testing::MatchResultListener* listener) const {
+  // Get all headers with matching keys.
+  std::vector<absl::string_view> values;
+  std::pair<std::string, std::vector<absl::string_view>*> context =
+      std::make_pair(key_.get(), &values);
+  Envoy::Http::HeaderMap::ConstIterateCb get_headers_cb = [](const Envoy::Http::HeaderEntry& header,
+                                                             void* context) {
+    auto* typed_context =
+        static_cast<std::pair<std::string, std::vector<absl::string_view>*>*>(context);
+    if (header.key().getStringView() == typed_context->first) {
+      typed_context->second->push_back(header.value().getStringView());
+    }
+    return Envoy::Http::HeaderMap::Iterate::Continue;
+  };
+  headers.iterate(get_headers_cb, &context);
+
+  if (values.empty()) {
+    *listener << "which has no '" << key_.get() << "' header";
+    return false;
+  } else if (values.size() > 1) {
+    *listener << "which has " << absl::StrCat(values.size()) << " '" << key_.get()
+              << "' headers, with values: " << absl::StrJoin(values, ", ");
```

Comment:
It's not clear to me why `StrCat` is used here. Can't you just say `"which has " << values.size() << " '"`etc? The StrJoin makes sense.

Reply:
Done.